### PR TITLE
test: replace Calendar part test with testing lib

### PR DESCRIPTION
**Pull request by lzm0x219**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [x] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    -       |
| 🇨🇳 Chinese |    -       |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/calendar/__tests__/__snapshots__/index.test.js.snap [hunk 1]
```diff
@@ -1,2833 +0,0 @@
...
-                </div>
-              </td>
-            </tr>
-          </tbody>
-        </table>
-      </div>
-    </div>
-  </div>
-</div>
-`;
```
Comment (afc163) [inline components/calendar/__tests__/__snapshots__/index.test.js.snap:2833]: 这个文件为啥会没掉

Comment (zombieJ) [reply components/calendar/__tests__/__snapshots__/index.test.js.snap:2833]: js 改成 tsx 了

File: components/calendar/__tests__/index.test.tsx [hunk 2]
```diff
@@ -121,19 +120,24 @@ describe('Calendar', () => {
 
   it('getDateRange should returns a disabledDate function', () => {
     const validRange: [Moment.Moment, Moment.Moment] = [Moment('2018-02-02'), Moment('2018-05-18')];
-    const wrapper = mount(<Calendar validRange={validRange} defaultValue={Moment('2018-02-02')} />);
-    const { disabledDate } = wrapper.find('PickerPanel').props() as any;
+    const disabledDate = jest.fn((x: Moment.Moment) => {
```
Comment (zombieJ) [inline components/calendar/__tests__/index.test.tsx:123]: 这个不对，需要测 Calendar 内部的 disabledDate，而不是写一个无关的。

File: components/calendar/__tests__/index.test.tsx [hunk 3]
```diff
@@ -121,19 +120,24 @@ describe('Calendar', () => {
...
+    });
+    render(<Calendar validRange={validRange} defaultValue={Moment('2018-02-02')} />);
     expect(disabledDate(Moment('2018-06-02'))).toBe(true);
     expect(disabledDate(Moment('2018-04-02'))).toBe(false);
   });
 
   it('validRange should work with disabledDate function', () => {
     const validRange: [Moment.Moment, Moment.Moment] = [Moment('2018-02-02'), Moment('2018-05-18')];
-    const wrapper = mount(
+    const disabledDate = jest.fn((x: Moment.Moment) => {
```
Comment (zombieJ) [inline components/calendar/__tests__/index.test.tsx:134]: 同上

File: components/calendar/__tests__/index.test.tsx [hunk 4]
```diff
@@ -143,46 +147,53 @@ describe('Calendar', () => {
 
   it('Calendar MonthSelect should display correct label', () => {
     const validRange: [Moment.Moment, Moment.Moment] = [Moment('2018-02-02'), Moment('2019-06-1')];
-    const wrapper = mount(<Calendar validRange={validRange} defaultValue={Moment('2019-01-01')} />);
-    expect(wrapper.render()).toMatchSnapshot();
+    const wrapper = render(
+      <Calendar validRange={validRange} defaultValue={Moment('2019-01-01')} />,
+    );
+    expect(wrapper.container).toMatchSnapshot();
```
Comment (zombieJ) [inline components/calendar/__tests__/index.test.tsx:153]: 比较一下子元素，否则会多一个额外的 div：
```tsx
expect(wrapper.container.innerHTML).toMatchSnapshot();
```

Comment (zombieJ) [reply components/calendar/__tests__/index.test.tsx:153]: 下面用到 toMatchSnapshot 的改动一样~~

File: components/calendar/__tests__/index.test.tsx (hunk 2 above)
Comment (0x219) [reply components/calendar/__tests__/index.test.tsx:123]: (o´ω`o)ﾉ 这里需要help

File: components/calendar/__tests__/index.test.tsx (hunk 4 above)
Comment (0x219) [reply components/calendar/__tests__/index.test.tsx:153]: 欧克

File: components/calendar/__tests__/__snapshots__/index.test.tsx.snap [hunk 5]
```diff
@@ -1,1892 +1,8 @@
...
...v class=\\"ant-picker-cell-inner ant-picker-calendar-date\\"><div class=\\"ant-picker-calendar-date-value\\">05</div><div class=\\"ant-picker-calendar-date-content\\"></div></div></td><td title=\\"2019-02-06\\" class=\\"ant-picker-cell\\"><div class=\\"ant-picker-cell-inner ant-picker-calendar-date\\"><div class=\\"ant-picker-calendar-date-value\\">06</div><div class=\\"ant-picker-calendar-date-content\\"></div></div></td><td title=\\"2019-02-07\\" class=\\"ant-picker-cell\\"><div class=\\"ant-picker-cell-inner ant-picker-calendar-date\\"><div class=\\"ant-picker-calendar-date-value\\">07</div><div class=\\"ant-picker-calendar-date-content\\"></div></div></td><td title=\\"2019-02-08\\" class=\\"ant-picker-cell\\"><div class=\\"ant-picker-cell-inner ant-picker-calendar-date\\"><div class=\\"ant-picker-calendar-date-value\\">08</div><div class=\\"ant-picker-calendar-date-content\\"></div></div></td><td title=\\"2019-02-09\\" class=\\"ant-picker-cell\\"><div class=\\"ant-picker-cell-inner ant-picker-calendar-date\\"><div class=\\"ant-picker-calendar-date-value\\">09</div><div class=\\"ant-picker-calendar-date-content\\"></div></div></td></tr></tbody></table></div></div></div></div>"`;
```
Comment (afc163) [inline components/calendar/__tests__/__snapshots__/index.test.tsx.snap:3]: 这个 snapshot 没换行，有点不可读。

Comment (0x219) [reply components/calendar/__tests__/__snapshots__/index.test.tsx.snap:3]: 好的 我瞅瞅

**Comment by afc163** [comment]
这个啥也没改怎么就合并了？

**Comment by yoyo837** [comment]
revert吗？/手动斜眼 [旺柴]

**Comment by yoyo837** [comment]
@0x219

**Comment by yoyo837** [comment]
Files changed 0  😺

**Pull request merged**